### PR TITLE
Task 6 (SNS and SQS)

### PR DESCRIPTION
**Pull request by otisorigin**
Primary scope Done:
1 - File serverless.yml contains configuration for catalogBatchProcess function
2 - File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
3 - File serverless.yml contains configuration for SQS catalogItemsQueue
4 - File serverless.yml contains configuration for SNS Topic createProductTopic and email subscription

CloudFront with application: https://d16eeo3sn01bhm.cloudfront.net/

### CODE REVIEW

File: product-service/src/lambdas/catalogBatchProcess.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        productResult.rows[0].id,
+        product.count,
+      ]);
+      sns.publish(
+        {
+          Subject: "Product created",
+          Message: JSON.stringify(product),
+          TopicArn: process.env.SNS_ARN,
+        },
+        () => {
```
Comment (IvanPresmytsky) [inline product-service/src/lambdas/catalogBatchProcess.ts:46]: It's a good idea to handle error here for the case when something goes wrong with the publishing a message.

File: product-service/src/lambdas/catalogBatchProcess.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+): Promise<void> => {
+  let client;
+  try {
+    console.log(event);
+    const products = event.Records.map(
+      ({ body }) => JSON.parse(body) as Product
+    ) as Product[];
+    console.log(products);
+    const sns = new AWS.SNS({ region: "eu-west-1" });
+    client = new Client((dbOptions as unknown) as ClientConfig);
```
Comment (IvanPresmytsky) [inline product-service/src/lambdas/catalogBatchProcess.ts:27]: The main idea for this lambda was to post a request with product data to the 'createProduct' lambda and then that lambda should write it to DB.

**Review by IvanPresmytsky — APPROVED** [review]
Good job!

File: product-service/src/lambdas/catalogBatchProcess.ts (hunk 2 above)
Comment (ArtiFisher) [reply product-service/src/lambdas/catalogBatchProcess.ts:27]: As far as I understood - it should be done in this lambda.
"Configure the SQS to trigger lambda catalogBatchProcess with 5 messages at once via batchSize property. The lambda function should iterate over all SQS messages **and create corresponding products in the products table**."

Comment (IvanPresmytsky) [reply product-service/src/lambdas/catalogBatchProcess.ts:27]: You might be right. There is some confusion between requirements, QA session, and QA document.

**Pull request merged**